src/masternodes/mn_checks.cpp

```diff
@@ -4457,9 +4451,8 @@ ResVal<uint256> ApplyAnchorRewardTxPlus(CCustomCSView &mnview,
             cbValues.begin()->second,
             anchorReward);
 
-    CTxDestination destination = finMsg.rewardKeyType == 1 ? CTxDestination(PKHash(finMsg.rewardKeyID))
-                                                           : CTxDestination(WitnessV0KeyHash(finMsg.rewardKeyID));
-    Require(tx.vout[1].scriptPubKey == GetScriptForDestination(destination), "anchor pay destination is incorrect");
+    CTxDestination destination = GetRewardDestinationOrDefaultFromKey(finMsg.rewardKeyType, finMsg.rewardKeyID);
+    Require(IsValidDestination(destination) && tx.vout[1].scriptPubKey == GetScriptForDestination(destination), "anchor pay destination is incorrect");
```

Comment:
We decided to remove the `Require` in the project so i think you can replace it with a simple if.

Reply:
I will leave the refactoring to remove all Require in a separate PR since that is not handled here yet.

Reply:
Resolved in commit 787927e20